src/Plugin/KeyInput/ApigeeAuthKeyInput.php

```diff
@@ -259,6 +260,21 @@ public function validateConfigurationForm(array &$form, FormStateInterface $form
    * {@inheritdoc}
    */
   public function processSubmittedKeyValue(FormStateInterface $form_state) {
+    $key_value = $form_state->get('key_value')['current'];
+    // Either null or an empty string.
+    if (empty($key_value)) {
+      // When "Test connection" reloads the page they are not yet processed.
+      // @see \Drupal\key\Form\KeyFormBase::createPluginFormState()
+      $key_input_plugin_form_state = clone $form_state;
+      $key_input_plugin_form_state->setValues($form_state->getUserInput()['key_input_settings']);
+      // @see \Drupal\key\Form\KeyFormBase::validateForm()
+      $key_input_processed_values = $form_state->getFormObject()->getEntity()->getKeyInput()->processSubmittedKeyValue($key_input_plugin_form_state);
+      $key_value = $key_input_processed_values['processed_submitted'];
+    }
+
+    // Could be an empty array.
+    $values = Json::decode($key_value);
+
```

Comment:
Thanks @minnur , this is looking good. However something around this section is giving me a white screen when checking the scenario of the module just installed (no credentials have ever been set). Could you try to replicate?

Reply:
@arlina-espinoza looks like the module is generating WSOD even without my changes :(

Reply:
@arlina-espinoza I think I got this working. Would you mind reviewing the changes?